packages/flutter_tools/lib/src/commands/analyze_base.dart

```diff
@@ -129,11 +134,12 @@ class PackageDependency {
   }
   bool get hasConflict => values.length > 1;
   bool get hasConflictAffectingFlutterRepo {
-    assert(globals.fs.path.isAbsolute(Cache.flutterRoot));
+    final String? flutterRoot = Cache.flutterRoot;
+    assert(flutterRoot != null && globals.fs.path.isAbsolute(flutterRoot));
     for (final List<String> targetSources in values.values) {
       for (final String source in targetSources) {
         assert(globals.fs.path.isAbsolute(source));
-        if (globals.fs.path.isWithin(Cache.flutterRoot, source)) {
+        if (flutterRoot != null && globals.fs.path.isWithin(flutterRoot, source)) {
```

Comment:
we assert that it's not null above, so presumably there's no need to check it again here